File: TrackIt.Services/Database/DailyIntake.cs
namespace TrackIt.Services.Database;

public partial class DailyIntake
{
	public int DailyIntakeId { get; set; }

	public int UserId { get; set; }

	public DateTime? Day { get; set; }

	public double? Calories { get; set; }

	public double? Carbs { get; set; }

	public double? Protein { get; set; }

	public double? Fat { get; set; }

	public double? Fiber { get; set; }

	public virtual GeneralUser User { get; set; } = null!;
}

File: UI/trackit_mobile/lib/models/DailyIntake/daily_intake.dart
import 'package:json_annotation/json_annotation.dart';
part 'daily_intake.g.dart';

@JsonSerializable()
class DailyIntake {
  int? dailyIntakeId;
  int? userId;
  DateTime? day;
  double? calories;
  double? carbs;
  double? protein;
  double? fat;

  DailyIntake(this.dailyIntakeId, this.userId, this.day, this.calories,
      this.carbs, this.protein, this.fat);

  factory DailyIntake.fromJson(Map<String, dynamic> json) =>
      _$DailyIntakeFromJson(json);

  Map<String, dynamic> toJson() => _$DailyIntakeToJson(this);
}
